chef/cookbooks/hyperv/recipes/openstack_install_ceilometer.rb

```diff
@@ -1,26 +1,46 @@
-raise if not node[:platform_family] == "windows"
+raise unless node[:platform_family] == "windows"
 
-cookbook_file "#{node[:cache_location]}#{node[:openstack][:ceilometer][:file]}" do
-  source node[:openstack][:ceilometer][:file]
-  not_if { ::File.exists?(node[:openstack][:ceilometer][:installed]) }
+component = node[:openstack][:ceilometer][:name]
+service = node[:service][:ceilometer][:name]
+
+installed_file = "#{node[:openstack][:location]}\\installed-#{component}"
+if File.exist? installed_file
+  Chef::Log.info("#{component} files already installed")
+  return
+end
+
+tarball = "#{component}-#{node[:openstack][:tarball_branch]}.tar.gz"
+cached_file = "#{node[:cache_location]}#{tarball}"
+cookbook_file cached_file do
+  source tarball
 end
 
-windows_batch "unzip_ceilometer" do
+windows_batch "unzip #{component}" do
   code <<-EOH
-  #{node[:sevenzip][:command]} x #{node[:cache_location]}#{node[:openstack][:ceilometer][:file]} -o#{node[:openstack][:location]} -r -y
-  #{node[:sevenzip][:command]} x #{node[:openstack][:location]}\\dist\\#{node[:openstack][:ceilometer][:name]}-#{node[:openstack][:ceilometer][:version]}.tar -o#{node[:openstack][:location]} -r -y
+  rmdir /S /Q #{node[:openstack][:location]}\\#{component}
+  #{node[:sevenzip][:command]} x #{cached_file} -o#{node[:openstack][:location]} -r -y
+  #{node[:sevenzip][:command]} x #{node[:openstack][:location]}\\dist\\#{component}-#{node[:openstack][:tarball_branch]}.tar -o#{node[:openstack][:location]} -r -y
   rmdir /S /Q #{node[:openstack][:location]}\\dist
-  ren #{node[:openstack][:location]}\\#{node[:openstack][:ceilometer][:name]}-#{node[:openstack][:ceilometer][:version]} #{node[:openstack][:ceilometer][:name]}
+  ren #{node[:openstack][:location]}\\#{component}-* #{component}
   EOH
-  not_if { ::File.exists?("#{node[:openstack][:location]}\\#{node[:openstack][:ceilometer][:name]}") }
 end
 
-powershell "install_ceilometer" do
+powershell "install #{component}" do
   code <<-EOH
   cd #{node[:openstack][:location]}
-  cd #{node[:openstack][:ceilometer][:name]}
+  cd #{component}
   $env:PBR_VERSION=Get-Content setup.cfg | Select-String -Pattern "version = " | %{$_ -replace "version = ", ""}
   #{node[:python][:command]} setup.py install
   EOH
-  not_if { ::File.exists?("#{node[:openstack][:ceilometer][:installed]}") }
+end
+
+utils_line "ensure correct python path in shebang in #{component}" do
+  file "#{node[:python][:scripts]}\\#{service}-script.py"
+  regexp /\A#!.*/
```

Comment:
Lint/AmbiguousRegexpLiteral: Ambiguous regexp literal. Parenthesize the method arguments if it's surely a regexp literal, or add a whitespace to the right of the `/` if it should be a division.